PYIC-8585: Discard async CRI queue message if no pending record

AmritSidhu opened the pull request:
## Proposed changes
### What changed

-  Discard async CRI messages with no pending record instead of retrying; update tests accordingly

Before:
<img width="600" height="778" alt="Screenshot 2025-08-20 at 11 18 20" src="https://github.com/user-attachments/assets/0e750e85-6094-4dea-a0d0-ea47eb969bd4" />

After:
<img width="600" height="565" alt="Screenshot 2025-08-20 at 11 18 03" src="https://github.com/user-attachments/assets/fc344909-7c97-46a9-9df6-e438477b3d6b" />

Expanded new log:
<img width="600" height="783" alt="Screenshot 2025-08-20 at 11 19 58" src="https://github.com/user-attachments/assets/64320924-2e32-4ea3-bdc0-954a9519da1c" />

### Why did it change

-  The retries are causing unnecessary processing, noise in logs and are filling up mobile’s DLQ, making it hard to identify genuine issues.

### Issue tracking
<!-- Jira ticket & other docs, like RFCs -->

- [PYIC-8585](https://govukverify.atlassian.net/browse/PYIC-8585)

## Checklists

- [ ] READMEs and documentation up-to-date
- [x] API/ unit/ contract tests have been written/ updated
- [x] No risk of exposure: PII, credentials, etc through logs/ code
- [ ] Production changes appropriately staged out


[PYIC-8585]: https://govukverify.atlassian.net/browse/PYIC-8585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8585: Discard async CRI queue message if no pending record

Discard async CRI messages with no pending record instead of retrying; update tests accordingly

blakeyp commented on lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/ProcessAsyncCriCredentialHandlerTest.java:
```diff
@@ -190,11 +190,11 @@ void shouldRejectValidUnexpectedVerifiableCredential() throws Exception {
 
     @Test
-    void shouldRejectValidUnsolicitedVerifiableCredential() throws Exception {
+    void shouldDiscardValidUnsolicitedVerifiableCredentialWithoutRetry() throws Exception {
```
Do we need both of these test cases? they look like they're doing the same thing 🧐

blakeyp commented on lambdas/process-async-cri-credential/src/main/java/uk/gov/di/ipv/core/processasynccricredential/ProcessAsyncCriCredentialHandler.java:
```diff
@@ -223,8 +223,9 @@ private void processSuccessAsyncCriResponse(SuccessAsyncCriResponse successAsync
         if (criResponseItem.isEmpty()) {
             LOGGER.error(
-                    LogHelper.buildLogMessage("No response item found given user id and state"));
-            throw new AsyncVerifiableCredentialException(UNEXPECTED_ASYNC_VERIFIABLE_CREDENTIAL);
```
Do we use this exception or error code anywhere else? if not we can bin it

AmritSidhu committed:
PYIC-8585: discard unsolicited success responses; remove unexpected-async-VC path & redundant test

Delete duplicate test
Remove handling for AsyncVerifiableCredentialException and the UNEXPECTED_ASYNC_VERIFIABLE_CREDENTIAL constant
